### PR TITLE
[SPARK-39562][SQL][TESTS] Make `hive-thriftserver` module unit tests pass in IPv6 env

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to make `hive-thrift server` module unit tests pass in IPv6 environment.

### Why are the changes needed?

To have IPv6 test coverage of Spark Thrift Server.

### Does this PR introduce _any_ user-facing change?

No. This is a test-only change.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Review by wangyum — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @wangyum !

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon  and @wangyum .
Merged to master.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala [hunk 1]
```diff
@@ -1189,6 +1189,7 @@ abstract class HiveThriftServer2TestBase extends SparkFunSuite with BeforeAndAft
   protected val startScript = "../../sbin/start-thriftserver.sh".split("/").mkString(File.separator)
   protected val stopScript = "../../sbin/stop-thriftserver.sh".split("/").mkString(File.separator)
 
+  val localhost = Utils.localHostNameForURI()
```
Comment (cloud-fan) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1192]: Shall we use `Utils.localCanonicalHostName` as https://github.com/apache/spark/pull/37377 suggested?

Comment (dongjoon-hyun) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1192]: Oh, did you hit any failure, @cloud-fan ? We can if we need. IIUC, `KafkaTestUtils` needs it previously due to MiniKDC `principal` string.

Comment (cloud-fan) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1192]: yea I found it when running this test in our internal test infra. Opened https://github.com/apache/spark/pull/37656

**Pull request closed without merging**